Source/AwsCommonRuntimeKit/mqtt/Mqtt5Packets.swift

```diff
@@ -179,63 +179,105 @@ public class PublishPacket: CStruct {
     func withCStruct<Result>(_ body: (aws_mqtt5_packet_publish_view) -> Result) -> Result {
         var raw_publish_view = aws_mqtt5_packet_publish_view()
 
-        raw_publish_view.qos = aws_mqtt5_qos(UInt32(qos.rawValue))
+        raw_publish_view.qos = self.qos.nativeValue
         raw_publish_view.retain = retain
         return topic.withByteCursor { topicCustor in
             raw_publish_view.topic =  topicCustor
             return withOptionalAWSByteCursorFromData(to: payload) { cByteCursor in
                 raw_publish_view.payload = cByteCursor
 
                 let _payloadFormatIndicatorInt: aws_mqtt5_payload_format_indicator? =
-                                                                    payloadFormatIndicator?.rawValue ?? nil
+                                                                    payloadFormatIndicator?.nativeValue ?? nil
```

Comment:
You don't need the ??. `?.` will return nill if it's nil. See https://docs.swift.org/swift-book/documentation/the-swift-programming-language/optionalchaining/